compiler/src/dotty/tools/dotc/transform/PickleQuotes.scala

```diff
@@ -135,16 +136,9 @@ class PickleQuotes extends MacroTransform {
       }
 
       /** Remove references to local types that will not be defined in this quote */
-      private def getTermHoleType(using Context) = new TypeMap() {
-        override def apply(tp: Type): Type = tp match
-          case tp @ TypeRef(NoPrefix, _) =>
-            // reference to term with a type defined in outer quote
-            getTypeHoleType(tp)
-          case tp @ TermRef(NoPrefix, _) =>
-            // widen term refs to terms defined in outer quote
-            apply(tp.widenTermRefExpr)
-          case tp =>
-            mapOver(tp)
+      private def getTermHoleType(using Context) = new AvoidMap {
+        def toAvoid(tp: NamedType): Boolean =
+          tp.prefix == NoPrefix // reference to terms and type defined in outer quote
```

Comment:
What about references defined in the quote itself, e.g.:
```scala
'{ class A; val x: A = new A; ... }
```
they have NoPrefix as a prefix, but they're not defined in an outer quote.

Reply:
I will probably need to move this into the Splicing phase where we know the set of definitions that are in the outer quote. I will open a separate PR. Should I revert that change in this PR? As far as I can see we are already avoiding those references.

Reply:
> R. Should I revert that change in this PR? 

Sure!